pennylane/ops/op_math/condition.py

```diff
@@ -666,17 +681,42 @@ def _get_cond_qfunc_prim():
     cond_prim.multiple_results = True
 
     @cond_prim.def_impl
-    def _(conditions, *args_and_consts, jaxpr_branches, n_consts_per_branch, n_args):
-
-        args = args_and_consts[:n_args]
-        consts_flat = args_and_consts[n_args:]
+    def _(*all_args, jaxpr_branches, n_consts_per_branch, n_args):
+        n_branches = len(jaxpr_branches)
+        conditions = all_args[:n_branches]
+        args = all_args[n_branches : n_branches + n_args]
+        consts_flat = all_args[n_branches + n_args :]
+
+        # Find predicates that use mid-circuit measurements. We don't check the last
+        # condition as that is always `True`.
+        mcm_conditions = [pred for pred in conditions[:-1] if isinstance(pred, MeasurementValue)]
+        if len(mcm_conditions) != 0:
+            if len(mcm_conditions) != len(conditions) - 1:
+                raise ConditionalTransformError(
+                    "Cannot use qml.cond with a combination of mid-circuit measurements "
+                    "and other classical conditions as predicates."

```

Comment:
Why is this? Don't `MeasurementValue` and `bool` play nicely together? :)

Reply:
See my comment above.